### PR TITLE
Added aws-sim provider

### PR DESCRIPTION
**Pull request by henryh2**
Added functionallity for an 'aws-sim' provider that can be used to simulate the AWS capacity blocks API.

### CODE REVIEW

**Comment by henryh2** [comment]
Potential items that should be considered for this MR:
- Exploring the edge cases for DescribeInstanceTopology endpoint, and how the API responds to incorrect input.
- Test to make sure multiple "pages" work correctly within the simulated client
- Add more test cases for the model structure
- Do more in-depth validation of model structure on import

File: pkg/providers/aws/provider_sim.go [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+				instTopo.InstanceType = &node.Type
+				instTopo.AvailabilityZone = &az
+				instTopo.ZoneId = &az
+				instTopo.GroupName = &pg
+				instTopo.CapacityBlockId = &node.CapacityBlock
+				var netLayers []string
+				for j := len(node.NetLayers) - 1; j >= 0; j-- {
+					netLayers = append(netLayers, node.NetLayers[j])
+				}
+				instTopo.NetworkNodes = netLayers
```
Comment (pkedy) [inline pkg/providers/aws/provider_sim.go]: Style nit

```go
				var netLayers []string
				for j := len(node.NetLayers) - 1; j >= 0; j-- {
					netLayers = append(netLayers, node.NetLayers[j])
				}
				instTopo := types.InstanceTopology{
					InstanceId: &instanceId,
					InstanceType: &node.Type,
					AvailabilityZone: &az,
					ZoneId: &az,
					GroupName: &pg,
					CapacityBlockId: &node.CapacityBlock,
					NetworkNodes = netLayers,
				}
```

File: pkg/providers/aws/provider_sim.go [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+	clientFactory := func(region string) (*Client, error) {
+		csp_model, err := models.NewModelFromFile(p.ModelPath)
+		if err != nil {
+			return nil, fmt.Errorf("unable to load model file for AWS simulation, %v", err)
+		}
+		simClient := SimClient{Model: csp_model}
+
+		return &Client{
+			EC2: simClient,
+		}, nil
```
Comment (pkedy) [inline pkg/providers/aws/provider_sim.go]: Nit: Since nothing is created scoped to `region`, you could move this code out to `LoaderSim` and simply have clientFactory return a singleton instance of `*Client`.

**Review by pkedy — APPROVED** [review]
Just a couple style nits.

**Comment by pkedy** [comment]
For a future PR: Since this is a simulated implementation, it should have a unit test.

File: pkg/providers/aws/provider_sim.go (hunk 1 above)
Comment (henryh2) [reply pkg/providers/aws/provider_sim.go]: Changed.

File: pkg/providers/aws/provider_sim.go (hunk 2 above)
Comment (henryh2) [reply pkg/providers/aws/provider_sim.go]: Also changed.

**Review by dmitsh — APPROVED** [review]
